src/vm/op_contract.rs

```diff
@@ -227,11 +227,8 @@ impl InstructionSet for ContractOp {
         }
         macro_rules! load_inputs {
             ($state_type:ident) => {{
-                if !context.prev_state.contains_key($state_type) {
-                    return ExecStep::Next;
-                }
                 let Some(prev_state) = context.prev_state.get($state_type) else {
-                    fail!()
+                    return ExecStep::Next;
```

Comment:
This should be fail!()
```suggestion
                    fail!()
```